components/micronova.rst

```diff
@@ -0,0 +1,201 @@
+    The MicroNova bords come in various flavours. This code is only tested on an ExtraFlame Ketty Evo 2.0 stove. The protocol is not
+    documented but has been reverse engineerd by others. See the links below for all the info that helped me.
+    The different sensors, buttons and stove switch may require specific **memory_location** and **memory_address** parameters that 
+    match your MicroNova specific board.
+
+    Also, switching your stove on or off can behave different on the various MicroNova flavours.
+
+    Use this integration at your own risk.
+
+Connecting your stove
+---------------------
+
+Most MicroNova based pellet stoves have a serial output. In most cases this output has 4 pins: GND, 5v, 20V and DATA.
+
+.. figure:: images/micronova_serial.png
+    :align: center
+    :width: 50.0%
+
+.. figure:: images/micronova_serial_layout.png
+    :align: center
+    :width: 50.0%
+
+You will have to build a simple circuit to interface with your stove. It is based on optocouplers for galvanic separation and logic 
+level shifting between 5v and 3V3.
+
+.. figure:: images/micronova_optocouplers.png
+    :align: center
+    :width: 100.0%
+
+    Optocoupler interface circuit (credit: philibertc)
+
+
+
+See the references below for all the details about te circuit.
+
+You can use the 5V output from the stove to power the ESP module, but you will have to put a voltage regulator in between to
+get 3v3.
+
+MicroNova configuration
+-----------------------
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    uart:
+      tx_pin: 5
+      rx_pin: 4
+      baud_rate: 1200
+      stop_bits: 2
+
+    micronova:
+      enable_rx_pin: 4
+      update_interval: 20s
+
+    text_sensor:
+      - platform: micronova
+        stove_state:
+          name: Stove status
+
+    number:
+      - platform: micronova
+        thermostat_temperature:
+          name: Thermostat temperature
+          step: 1
+      - platform: micronova
+        power_level:
+          name: Thermostat temperature
+
+    sensor:
+      - platform: micronova
+        room_temperature:
+          name: Room temperature
+        fumes_temperature:
+          name: Fumes temperature
+        water_temperature:
+          name: Water temperature
+          # every sensor can specify memory parameters
+          memory_location: 0x00
+          memory_address: 0x3B
+        water_pressure:
+          name: Water pressure
+        stove_power:
+          name: Stove power level
+        fan_speed:
+          fan_rpm_offset: 240
+          name: Fan RPM
+        memory_address_sensor:
+          memory_location: 0x20
+          memory_address: 0x7d
+          name: Custom Address sensor
+
+    switch:
+      - platform: micronova
+        stove:
+          name: Stove on/off switch
+
+    button:
+      - platform: micronova
+        custom_button:
+          name: Custom button
+          memory_location: 0x20
+          memory_address: 0x7d
+          memory_data: 0x08
```

Comment:
Please move all of these into their own code block under their respective headings below.

Reply:
Done